### ISSUE TITLE
file: wrong dimensions for GIF images

### ISSUE THREAD

**Issue by rprichard** [landley/toybox#225]
The second dimension (height) of a GIF image reported by `toybox file` is wrong.

[Bug.gif](https://user-images.githubusercontent.com/1572855/84723139-cb7a2d80-af39-11ea-8c03-d21e79359798.gif)

```
$ /x/toybox/toybox file Bug.gif 
Bug.gif: GIF image data, 32 x 0
$ file Bug.gif 
Bug.gif: GIF image data, version 89a, 32 x 32
```

On the files in my /usr/share directory, sometimes toybox reports 0 or some other large value:
```
/usr/share/apache2/icons/small/key.gif: GIF image data, 16 x 0
/usr/share/apache2/icons/small/key.gif: GIF image data, version 89a, 16 x 16

/usr/share/apache2/icons/small/folder2.gif: GIF image data, 16 x 49344
/usr/share/apache2/icons/small/folder2.gif: GIF image data, version 89a, 16 x 16

/usr/share/apache2/icons/small/movie.gif: GIF image data, 16 x 14649
/usr/share/apache2/icons/small/movie.gif: GIF image data, version 89a, 16 x 16

/usr/share/apache2/icons/ball.red.gif: GIF image data, 20 x 65535
/usr/share/apache2/icons/ball.red.gif: GIF image data, version 89a, 20 x 22

/usr/share/apache2/icons/apache_pb.gif: GIF image data, 260 x 53934
/usr/share/apache2/icons/apache_pb.gif: GIF image data, version 89a, 260 x 30
```
The dimensions for PNG files seem OK.

**Comment by enh-google** [comment]
works for me with your example and ToT toybox:
```
$ ./toybox file /tmp/bug.gif 
/tmp/bug.gif: GIF image data, 32 x 32
```

i tried an asan build too, no difference.

i also added the "version %s, " bit (i'll send the patch to the list) and then did:
```
/tmp/toybox-gif$ locate .gif | xargs file > /tmp/a.txt
/tmp/toybox-gif$ locate .gif | xargs ./toybox file > /tmp/b.txt
/tmp/toybox-gif$ diff -u /tmp/a.txt /tmp/b.txt
/tmp/toybox-gif$ 
```

**Comment by enh-google** [comment]
ah: rob already  fixed the bug with 718d73c023b765f1434edb5def188847b4cc044a  but didn't mention this bug in the commit message, so github didn't link the two.
